pkg/apis/awsproviderconfig/v1alpha1/register_test.go

```diff
@@ -43,10 +47,22 @@ func TestEncodeAndDecodeProviderStatus(t *testing.T) {
 		t.Error(err)
 	}
 
-	providerStatusDecoded := &AWSMachineProviderStatus{}
-	codec.DecodeProviderStatus(providerStatusEncoded, providerStatusDecoded)
-	if !reflect.DeepEqual(providerStatus, providerStatusDecoded) {
-		t.Errorf("failed EncodeProviderStatus/DecodeProviderStatus. Expected: %+v, got: %+v", providerStatus, providerStatusDecoded)
+	// without deep copy
+	{
```

Comment:
Because of the blocks I would be inclined to create two very obvious function names:

    TestDecodeProviderStatus()
    TestDecodeProviderStatusWithDeepCopy()

Reply:
I did not want to duplicate the code just because of few additional lines.